src/App.test.tsx

```diff
@@ -1,21 +1,80 @@
 import React from "react";
-import { render, screen, waitForElement } from "@testing-library/react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
 import App from "./App";
 import { StoreProvider } from "easy-peasy";
 import store from "./state/store";
 import { ThemeProvider } from "theme-ui";
 import theme from "./style/theme";
+import { act } from "react-dom/test-utils";
 
 window.scrollTo = jest.fn();
 
-test("renders claim element", () => {
-  const { getByText } = render(
-    <StoreProvider store={store}>
-      <ThemeProvider theme={theme}>
-        <App />
-      </ThemeProvider>
-    </StoreProvider>
-  );
-  const claimElement = getByText(/Offene Datenplattform für IoT-Projekte/i);
-  expect(claimElement).toBeInTheDocument();
+describe("App", () => {
+  test("initially renders claim element", () => {
+    render(
+      <StoreProvider store={store}>
+        <ThemeProvider theme={theme}>
+          <App />
+        </ThemeProvider>
+      </StoreProvider>
+    );
+
+    const claimElement = screen.getByText(
+      /Offene Datenplattform für IoT-Projekte/i
+    );
+    expect(claimElement).toBeInTheDocument();
+  });
+
+  test("displays headings of projects from API", async () => {
+    render(
+      <StoreProvider store={store}>
+        <ThemeProvider theme={theme}>
+          <App />
+        </ThemeProvider>
+      </StoreProvider>
+    );
+
+    const projectAHeading = await screen.findByText(/Test project A/i);
+    const projectBHeading = await screen.findByText(/Test project B/i);
+    expect(projectAHeading).toBeInTheDocument();
+    expect(projectBHeading).toBeInTheDocument();
+  });
+
+  test("displays cookie disclaimer", () => {
+    render(
+      <StoreProvider store={store}>
+        <ThemeProvider theme={theme}>
+          <App />
+        </ThemeProvider>
+      </StoreProvider>
+    );
+
+    const cookieBanner = screen.getByText(/Diese Webseite verwendet Cookies/i);
+    expect(cookieBanner).toBeInTheDocument();
+  });
+
+  test("allows navigating to project page", async () => {
+    await act(async () => {
+      render(
+        <StoreProvider store={store}>
+          <ThemeProvider theme={theme}>
+            <App />
+          </ThemeProvider>
+        </StoreProvider>
+      );
+
+      await act(async () => {
```

Comment:
@dnsos did you get the "your component is not wrapped in act" error? See https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning

Reply:
Yes, exactly. I came across this article, thanks for the recommendation. Will take a look at it!